setup-gcloud/README.md

```diff
@@ -44,3 +44,44 @@ jobs:
           docker build -t $IMAGE .
           docker push $IMAGE
 ```
+
+#### Make the action run faster
+
+This example will download the `gcloud` CLI and cache it for further use. The CLI is used to authenticate docker to later build and push an image to GCR.
+
+```yaml
+on: push
+
+jobs:
+  staging:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v1
+
+      - name: Get gcloud latest version
+        run: |
+          url="https://dl.google.com/dl/cloudsdk/channels/rapid/components-2.json"
+          version=$(curl -s "$url" | grep -o '"version": "[^"]*' | cut -d'"' -f4)
+          echo "gcloud_version=$version" >> $GITHUB_ENV
+
+      - name: Cache Gcloud packages
+        uses: actions/cache@v3
+        id: gcloud-cache
+        with:
+          key: ${{ runner.os }}-cache-gcloud-${{ env.gcloud_version }}
+          path: |
+            /opt/hostedtoolcache/gcloud/${{ env.gcloud_version }}
+
+      - uses: extenda/actions/setup-gcloud@v0
+        id: gcloud
+        with:
+          service-account-key: ${{ secrets.GCLOUD_AUTH_STAGING }}
+          cache: steps.gcloud-cache.outputs.cache-hit
```

Comment:
Do we need to manually get the latest and declare the cache? Would it be possible to use the cache toolkit and hide all of these?

Or as an alternative convert setup-cloud to a composite action that setup the cache internally?